src/common/Utilities.py

```diff
@@ -463,3 +463,29 @@ def str_to_int_list(str_list):
             int_item = list(range(strt, stop + 1))
         int_list = int_list + int_item
     return int_list
+
+def dot_product_test(operator, num_tests = 20, max_err = 10e-5):
+    '''
+    Test if a given operator is adjoint.
+    The operator needs to have been already set_up() with valid AcquisitionModel and ImageData objects.
+    The operator needs to have methods direct() and adjoint() implemented
+    '''
+    # Get image and data to know the sizes of each operation.
+    image=operator.domain_geometry()
+    image.fill(1.0)
+    data=operator.direct(image) # Is there a better way than this? Similar to .domain_geomerty() ?
+    for __ in range(num_tests):
+        # generate random data
+        x0 = numpy.random.rand(*image.shape) * 10**numpy.random.randint(-2, 3)
+        y0 = numpy.random.rand(*data.shape ) * 10**numpy.random.randint(-2, 3)
```

Comment:
I guess will have to find out if it needs to be complex or not. Can you get the data-type from `image` and `data`

Reply:
I suspect I can check against `numpy` types here, for complex? You don't have your own implementation for complex, right? I also do not see any method/attribute that tells me if its complex or not.